### ISSUE TITLE
react-navigation returns "RNSScreen was not found in the UIManager" on RNW 0.64

### ISSUE THREAD

**Issue by qmatteoq** [microsoft/react-native-windows#7565]
## Environment
Run the following in your terminal and copy the results here.
1. `npx react-native --version`:

5.0.1-alpha.2

2. `npx react-native info`:

```
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 3.30 GB / 15.60 GB
  Binaries:
    Node: 12.9.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.10.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.14393.0, 10.0.15063.0, 10.0.17134.0, 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.9.31129.286 (Visual Studio Enterprise 2019), 16.10.31025.218 (Visual Studio Enterprise 2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.0 => 0.64.0
    react-native-windows: ^0.64.0-0 => 0.64.3
  npmGlobalPackages:
    *react-native*: Not Found
```

- Target Platform Version(s): 10.0.18362
- Target Device(s): Desktop
- Visual Studio Version: Visual Studio 2019 16.9.3
- Build Configuration: Debug

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create a new React Native 0.64 project
2. Follow [the instructions](https://reactnavigation.org/docs/getting-started) to add react-navigation to the application
3. Add a basic stack navigation, like in the following sample:

```
import { NavigationContainer, StackActions } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

const HomeScreen = () => {
  return(
    <View>
      <Text>I'm the home screen</Text>
    </View>
  );
}

const App: () => React$Node = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );

```

## Expected Results
The application launches without errors. The application, instead, returns the following error:

```
folly::toJson: JSON object value was a NaN or INF
```

In the Chromium debugger, you can see a more detailed error:

```
C:\Users\mpagani\Source\React\BadgerMeterV64\node_modules\react-native-windows\Libraries\Core\ExceptionsManager.js:180 Invariant Violation: requireNativeComponent: "RNSScreen" was not found in the UIManager.
```

The same exact code, instead, works fine if the application is based on React Native for Windows 0.63.

**Comment by chrisglein** [comment]
Duplicate of #7566

@qmatteoq this looks like a dupe agree?

**Comment by qmatteoq** [comment]
Yes. My only concern is that the other issue has been closed with the message "react-native-screens isn't yet supported on Windows". While this is true, if you add to the Windows project an older version of react-native-screens (2.x) everything works fine also 0.64.

**Comment by kacperkapusciak** [comment]
Hi @qmatteoq
This issue should be now fixed in `react-native-screens` [version 3.1.1](https://github.com/software-mansion/react-native-screens/releases/tag/3.1.1)
Could you upgrade the package to the newest version and see if this helps?

**Comment by qmatteoq** [comment]
Thanks for the heads up! Yes, I confirm version 3.1.1 works just fine. I upgraded my RNW 0.64 project and the error is now gone. Thanks!
